src/index.js

```diff
@@ -110,13 +110,16 @@ async function run() {
         newVersion = versioning[0].newVersion
       }
 
-      const gitTag = `${tagPrefix}${newVersion}`
+      let gitTag = `${tagPrefix}${newVersion}`
 
       if (preChangelogGeneration) {
-        newVersion = await require(path.resolve(preChangelogGeneration)).preChangelogGeneration({
+        const newVersionAndTag = await require(path.resolve(preChangelogGeneration)).preChangelogGeneration({
           tag: gitTag,
           version: newVersion,
         })
+
+        gitTag = newVersionAndTag.tag
+        newVersion = newVersionAndTag.version
```

Comment:
Can we add checks like this to be sure?

```suggestion
        if(newVersionAndTag.version) {
            newVersion = newVersionAndTag.version
        }
```

Reply:
Yeah, sounds reasonable. Also there is some issue with running new test, will fix that as well.